packages/server-core/src/user/identity-provider/identity-provider.test.ts

```diff
@@ -69,37 +70,61 @@ describe('identity-provider service', () => {
       {
         type,
         token,
-        password
+        password,
+        userId
       },
       {}
     )
-    providers.push(item)
 
     assert.equal(item.type, type)
     assert.equal(item.token, token)
     assert.ok(item.userId)
   })
 
   it('should find identity providers', async () => {
-    for (const provider of providers) {
-      const item = await app.service('identity-provider').find({
-        query: {
-          userId: provider.userId
-        }
-      })
-
-      assert.ok(item, 'Identity provider item is found')
-    }
+    const item = await app.service('identity-provider').find({
+      query: {
+        userId
+      }
+    })
+
+    assert.ok(item, 'Identity provider item is found')
```

Comment:
If we're going to make all of these providers on the same user in this test, we should be testing that there are three identity providers in what's returned from identity-provider.find, just to make sure it's correct.

Reply:
Added this change!